tests/upgrades/scenario_workers.py

```diff
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+
+import pytest
+from automation_tools.satellite6.hammer import set_hammer_config
+from fabric.api import env
+
+from robottelo.config import configure_airgun
+from robottelo.config import configure_nailgun
+from robottelo.config import settings
+
+
+_json_file = 'upgrade_workers.json'
+
+
+def save_worker_hostname(test_name):
+    data = {}
+    json_file = Path(_json_file)
+    if json_file.exists():
+        data = json.loads(json_file.read_text())
+    # Removing the parameter name from test name before save
+    test_name = test_name.split('[')[0] if '[' in test_name else test_name
+    data.update({test_name: settings.server.hostname})
+    json_file.write_text(json.dumps(data))
+
+
+@pytest.fixture(scope='session')
+def shared_workers():
+    json_file = Path(_json_file)
+    if json_file.exists():
+        with open(_json_file) as rjson:
+            return json.load(rjson)
```

Comment:
```suggestion
        return json.loads(json_file.read_text())
```
same pattern as above